libraries/chain/include/eosio/chain/trace.hpp

```diff
@@ -19,6 +19,30 @@ namespace eosio::chain {
    struct transaction_trace;
    using transaction_trace_ptr = std::shared_ptr<transaction_trace>;
 
+   struct sync_call_trace {
+      sync_call_trace(uint32_t sender_ordinal, account_name sender, account_name receiver, uint64_t flags, std::span<const char> data)
+         : sender_ordinal(sender_ordinal)
+         , sender(sender)
+         , receiver(receiver)
+         , flags(flags)
+         , data(data.begin(), data.end())
+      {
+      }
+
+      uint32_t                      ordinal = 1;
+      const uint32_t                sender_ordinal = 0;
+      const account_name            sender;
```

Comment:
Is sender redundant information since you have `sender_ordinal`?

Reply:
I put `sender` here for simple reference. But it can be found from `sender_ordinal`. Will remove it.

Reply:
https://github.com/AntelopeIO/spring/pull/1333/commits/3038697fe5791145081852aa7a6103a2c83c8be0